crates/polars-pipe/src/pipeline/dispatcher.rs

```diff
@@ -330,89 +331,6 @@ impl PipeLine {
         self.sources.push(src);
     }
 
-    fn run_pipeline_no_finalize(
-        &mut self,
-        ec: &PExecutionContext,
-        pipeline_q: Rc<RefCell<VecDeque<PipeLine>>>,
-    ) -> PolarsResult<(u32, Box<dyn Sink>)> {
-        let mut out = None;
-        let mut operator_start = 0;
-        let last_i = self.sinks.len() - 1;
-
-        // for unions we typically first want to push all pipelines
-        // into the union sink before we call `finalize`
-        // however if the sink is finished early, (for instance a `head`)
-        // we don't want to run the rest of the pipelines and we finalize early
-        let mut sink_finished = false;
-
-        for (i, (operator_end, shared_count, mut sink)) in
-            std::mem::take(&mut self.sinks).into_iter().enumerate()
-        {
-            for src in &mut std::mem::take(&mut self.sources) {
-                let mut next_batches = src.get_batches(ec)?;
-
-                while let SourceResult::GotMoreData(chunks) = next_batches {
-                    let (sink_result, next_batches2) = self.par_process_chunks(
-                        chunks,
-                        &mut sink,
-                        ec,
-                        operator_start,
-                        operator_end,
-                        src,
-                    )?;
-                    next_batches = next_batches2;
-
-                    if let Some(SinkResult::Finished) = sink_result {
-                        sink_finished = true;
-                        break;
```

Comment:
I found this a bit strange—`sink_finished` is defined outside the "main" `for` loop, but here we only break the inner `while let` loop. Is that intentional?

Reply:
We should break the outer loop indeed.